packages/nimble-components/src/select/index.ts

```diff
@@ -736,6 +746,7 @@ export class Select
             this.setActiveOption(this.selectedIndex);
         }
         this.updateValue();
+        this.observeSelectedOptionTextContent();
```

Comment:
Can you talk about the decision to put this call in `selectedIndexChanged()` rather than, say `updateValue()`? It looks like `updateValue()` is called in a bunch of other scenarios like slotted options changing and clearing the selection. Should we stop observing immediately in those cases too?

Reply:
Conceptually, the idea is that we just want to react to the text content changing of the current selected option. This only needs to be changed (i.e. we need to observe a new option) when the actual selection changes, not when the display of the current selected option changes.

Reply:
While the corner-case you mention may have merit, to answer your larger question, I don't think this is a concern. The mutation observer doesn't react to synchronous changes to the DOM. Note that the only time calling `updateValue` results in the observed option becoming no longer selected is if the `selectedIndex` has been changed to -1, in which case there will be no option to observe, and the observer will be disconnected from the previous option (again before it ever bothers trying to update the display value, and even if it did it would be innocuous).

All that said, because of the corner-case you mention, I think calling `observerSelectedOptionTextContent` from within `updateValue` is a valid solution, and I'm not concerned with calling that method unnecessarily at times.

Reply:
I've updated the implementation and added a test to cover the corner-case you mention (failed without the change).